components/sync15/src/util.rs

```diff
@@ -42,9 +42,14 @@ impl From<i64> for ServerTimestamp {
 }
 
 impl From<f64> for ServerTimestamp {
-    #[inline]
     fn from(ts: f64) -> Self {
-        ServerTimestamp((ts * 1000.0).round() as i64)
+        let rf = (ts * 1000.0).round();
+        if !rf.is_finite() || rf < 0.0 || rf >= i64::max_value() as f64 {
+            log::error!("Illegal timestamp: {}", ts);
+            ServerTimestamp(0)
```

Comment:
What would previously happen with such values? I guess I'm wondering if there's any unforseen consequences of defaulting to zero in places where we weren't previously.  Or if this is a server-generated timestamp, can we assume it's generally well-behaved and this is just for completeness?

Reply:
Previously the compiler was free to miscompile things. In practice it might trap, you might get a random value, but in general LLVM assumes it absolutely cannot happen. If it could prove it can happen, then it's free to arbitrarily miscompile your program, more or less. Undefined behavior is bad news.